demo/full/bxf-parser2/src/parser/index.js

```diff
@@ -0,0 +1,38 @@
+import reduceChildren from "./reduce_children";
+import parseSchedule from "./schedule";
+
+/**
+ * Parses BXF XML (string format)
+ * and reduces it into a JSON.
+ * @param {string} xmlString
+ */
+export default function BXFParser(xmlString) {
+
+  const xmlDoc = new DOMParser().parseFromString(xmlString, "text/xml");
+  const bxf = xmlDoc
+    .childNodes[0].childNodes[1].childNodes[1].childNodes[1].childNodes[3].childNodes[1];
+
+  const bxfData = reduceChildren(bxf, (data, name, node) => {
+    switch (name.toLowerCase()) {
+      case "ns1:bxfdata":
+        data = parseBXFData(node);
+    }
+    return data;
+  }, undefined);
+
+  /**
+   * Extract schedule(s) from bxd Data, and parse it.
+   * @param {Element} bxfData
+   */
+  function parseBXFData(bxfData) {
+    return reduceChildren(bxfData, (res, name, node) => {
+      switch (name.toLowerCase()) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: "switch" statements should have at least 3 "case" clauses ([javascript:S1301](https://sonarcloud.io/organizations/rx-player/rules#rule_key=javascript%3AS1301))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=366&issues=AWbt7Z3jW3Y69bz5aGW-&open=AWbt7Z3jW3Y69bz5aGW-)

[ISSUE_KEY]: AWbt7Z3jW3Y69bz5aGW-